graphbolt/src/csc_sampling_graph.cc

```diff
@@ -120,45 +120,42 @@ c10::intrusive_ptr<SampledSubgraph> CSCSamplingGraph::InSubgraph(
           : torch::nullopt);
 }
 
+static c10::intrusive_ptr<CSCSamplingGraph> BuildGraphFromSharedMemoryTensors(
+    SharedMemoryTensors&& shared_memory_tensors) {
+  const auto& tensors = std::get<2>(shared_memory_tensors);
+  return c10::make_intrusive<CSCSamplingGraph>(
+      tensors[0], tensors[1],
+      tensors.size() > 2 ? torch::optional<torch::Tensor>{tensors[2]}
+                         : torch::nullopt,
+      tensors.size() > 3 ? torch::optional<torch::Tensor>{tensors[3]}
+                         : torch::nullopt,
+      std::move(std::get<0>(shared_memory_tensors)),
+      std::move(std::get<1>(shared_memory_tensors)));
+}
+
 c10::intrusive_ptr<CSCSamplingGraph> CSCSamplingGraph::CopyToSharedMemory(
     const std::string& shared_memory_name) {
-  auto shm_name_with_prefix = std::string("graphbolt_") + shared_memory_name;
   std::vector<torch::Tensor> tensors = {indptr_, indices_};
+  TORCH_CHECK(
```

Comment:
@frozenbugs I add a check here to avoid the case when the node_type_offset_ is None and type_per_edge_ is not None.

Reply:
Why we decided to not support a graph with type_per_edge but not node_type_offset?